pkg/virt-controller/watch/snapshot/restore.go

```diff
@@ -523,6 +537,8 @@ func (t *vmRestoreTarget) Reconcile() (bool, error) {
 			Spec:   *snapshotVM.Spec.DeepCopy(),
 			Status: kubevirtv1.VirtualMachineStatus{},
 		}
+
+		stripIdentityInfo(newVM)
```

Comment:
I think this is problematic. Let me give you some context first:

In order to [implement the cloning API](https://github.com/kubevirt/kubevirt/pull/7336) I first needed to [adapt](https://github.com/kubevirt/kubevirt/pull/7318) restore/snapshot controllers. The two are supposed to work in the following way:
* The snapshot / restore controllers can restore a snapshot into a brand new VM (if it doesn't exist).
* When this is being done, the restore creator can provide a set of patches (through the restore's spec).
  * The restore mechanism does not know how to strip a VM's identity. All it can do is get patches through spec and apply them on the target VM (offline, before it's created). This keeps a clear separation of concerns between the restore and clone APIs.
* The clone API is responsible to provide the right patches so the identity would be stripped properly.

What's needed to be done here is to add the proper patch into restore's functional tests. This way no identity stripping logic resides in the restore controller.

But maybe this can be done in a follow-up PR.